mongo/dataset_store.go

```diff
@@ -271,14 +272,21 @@ func (m *Mongo) UpdateDataset(id string, dataset *models.Dataset) (err error) {
 	s := m.Session.Copy()
 	defer s.Close()
 
-	updates := createDatasetUpdateQuery(dataset)
-	err = s.DB(m.Database).C("datasets").UpdateId(id, bson.M{"$set": updates, "$setOnInsert": bson.M{"next.last_updated": time.Now()}})
+	updates := createDatasetUpdateQuery(id, dataset)
+	if err = s.DB(m.Database).C("datasets").UpdateId(id, bson.M{"$set": updates, "$setOnInsert": bson.M{"next.last_updated": time.Now()}}); err != nil {
```

Comment:
consider pulling out the bson.M call onto a seperate line to reduce the length of this